applications/smart-distancing/eval/pascalvoc.py

```diff
@@ -0,0 +1,382 @@
+        fh1 = open(f, "r")
+        for line in fh1:
+            line = line.replace("\n", "")
+            if line.replace(" ", "") == "":
+                continue
+            splitLine = line.split(" ")
+            if isGT:
+                # idClass = int(splitLine[0]) #class
+                idClass = splitLine[0]  # class
+                x = float(splitLine[1])
+                y = float(splitLine[2])
+                w = float(splitLine[3])
+                h = float(splitLine[4])
+                bb = BoundingBox(
+                    nameOfImage,
+                    idClass,
+                    x,
+                    y,
+                    w,
+                    h,
+                    coordType,
+                    imgSize,
+                    BBType.GroundTruth,
+                    format=bbFormat,
+                )
+            else:
+                # idClass = int(splitLine[0]) #class
+                idClass = splitLine[0]  # class
+                confidence = float(splitLine[1])
+                x = float(splitLine[2])
+                y = float(splitLine[3])
+                w = float(splitLine[4])
+                h = float(splitLine[5])
+                bb = BoundingBox(
+                    nameOfImage,
+                    idClass,
+                    x,
+                    y,
+                    w,
+                    h,
+                    coordType,
+                    imgSize,
+                    BBType.Detected,
+                    confidence,
+                    format=bbFormat,
+                )
+            allBoundingBoxes.addBoundingBox(bb)
+            if idClass not in allClasses:
+                allClasses.append(idClass)
+        fh1.close()
+    return allBoundingBoxes, allClasses
+
+
+# Get current path to set default folders
+currentPath = os.path.dirname(os.path.abspath(__file__))
+
+VERSION = "0.1 (beta)"
+
+parser = argparse.ArgumentParser(
+    prog="Object Detection Metrics - Pascal VOC",
+    description="This project applies the most popular metrics used to evaluate object detection "
+    "algorithms.\nThe current implemention runs the Pascal VOC metrics.\nFor further references, "
+    "please check:\nhttps://github.com/rafaelpadilla/Object-Detection-Metrics",
+    epilog="Developed by: Rafael Padilla (rafael.padilla@smt.ufrj.br)",
+)
+# formatter_class=RawTextHelpFormatter)
+parser.add_argument("-v", "--version", action="version", version="%(prog)s " + VERSION)
+# Positional arguments
+# Mandatory
+parser.add_argument(
+    "-gt",
+    "--gtfolder",
+    dest="gtFolder",
+    default=os.path.join(currentPath, "groundtruth"),
+    metavar="",
+    help="folder containing your ground truth bounding boxes",
+)
+parser.add_argument(
+    "-det",
+    "--detfolder",
+    dest="detFolder",
+    default=os.path.join(currentPath, "detections_results"),
+    metavar="",
+    help="folder containing your detected bounding boxes",
+)
+# Optional
+parser.add_argument(
```

Comment:
if it was possible to evaluate for multiple IOU thresholds it would be great.

Reply:
Plan to have it at the next version of the evaluator in which the evaluator would be a separate docker file...